cairo/ethereum/utils/numeric.cairo

```diff
@@ -332,22 +333,47 @@ func Uint_bit_length{range_check_ptr}(value: Uint) -> felt {
 
     assert_le(bit_length, 252);
     let lower_bound = pow2(bit_length - 1);
-    assert_le_felt(lower_bound, value.value);
+    assert_le_felt(lower_bound, value);
     if (bit_length == 252) {
         return bit_length;
     }
     let upper_bound = pow2(bit_length);
-    assert_le_felt(value.value + 1, upper_bound);
+    assert_le_felt(value + 1, upper_bound);
 
     return bit_length;
 }
 
+func U256_bit_length{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(value: U256) -> felt {
+    alloc_locals;
+
+    tempvar u256_zero = U256(new U256Struct(low=0, high=0));
+    let is_zero_value = U256__eq__(value, u256_zero);
+    if (is_zero_value.value != 0) {
+        return 0;
+    }
+
+    let high_is_zero = is_zero(value.value.high);
+    if (high_is_zero == 0) {
+        let high_bit_length = _bit_length(value.value.high);
+        return high_bit_length + 128;
+    } else {
+        let low_bit_length = _bit_length(value.value.low);
+        return low_bit_length;
+    }
```

Comment:
```suggestion
    }
    return _bit_length(value.value.low);
    }
```

Reply:
it would require to have the same implicit arguments
Hence adding bitwise to _bit_length but no need for it

Reply:
ok 👌